### PR TITLE
05 redux basic

### PR DESCRIPTION
**Pull request by DzmitryI**

### CODE REVIEW

**Comment by vladislavkovaliov** [comment]
no issues

File: homeworks/05-redux-basic/redux-basic/.gitignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# See https://help.github.com/articles/ignoring-files/ for more about ignoring files.
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/.gitignore:1]: duplicate folder structure:
`05-redux-basic`
`redux-basic`

Please move all files to `05-redux-basic` and remove sub-folder `redux-basic`

File: homeworks/05-redux-basic/redux-basic/src/App.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
+.App {
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/App.css:1]: Please review the styles. Are they really used?

File: homeworks/05-redux-basic/redux-basic/src/App.test.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import App from './App';
+
+test('renders learn react link', () => {
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/App.test.tsx:5]: Is this test file used at all?

File: homeworks/05-redux-basic/redux-basic/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import './App.css';
+
+function App() {
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/App.tsx:4]: Inconsistent code indentation across files

File: homeworks/05-redux-basic/redux-basic/src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import {balance} from '../src/redux/reducers/balanceReducer'
+import {types} from '../src/redux/actions/actionsTypes'
+
+
+declare global {
+  interface Window {
+    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
+  }
+}
+const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/index.tsx:15]: Store creation should be separated in different file and then imported in App to wrap React components with store provider.

File: homeworks/05-redux-basic/redux-basic/src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+const store = createStore(balance, undefined, composeEnhancers());
+
+store.dispatch({ type: types.UPDATE_BALANCE, payload: 1000.0 });
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/index.tsx:18]: App component is not a test file to run dispatch actions.

File: homeworks/05-redux-basic/redux-basic/src/redux/actions/actionsTypes.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export enum types {
+	UPDATE_BALANCE = 'updateBalance',
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/redux/actions/actionsTypes.ts:2]: The action values should be in UPPER CASE

File: homeworks/05-redux-basic/redux-basic/src/redux/actions/actionsTypes.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+export enum types {
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/redux/actions/actionsTypes.ts:1]: types is not a good name for enum.
I suggest using `BalanceActionType`

File: homeworks/05-redux-basic/redux-basic/src/redux/reducers/balanceReducer.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import {types} from "../actions/actionsTypes";
+import {ActionTypes} from "../actions/balanceActions";
+
+export const balance = (state: number | undefined = 0, action: ActionTypes) => {
```
Comment (avilde) [inline homeworks/05-redux-basic/redux-basic/src/redux/reducers/balanceReducer.ts:4]: `balance` should be renamed to `balanceReducer`.

**Review by avilde — COMMENTED** [review]
## Review

# Summary
* file structure could be improved my removing unused files
* code indentation should be consistent 
* application is barely finished, UI is not fully done (no buttons, balance controls at all)
* messy code in `App.tsx` - store creation should be separated and dispatch actions should not be called in React component without a reason (either write unit tests or map UI controls to dispatch actions)
* second home work part is not done at all (Counter and combine reducers)

**Pull request merged**